src/renderer/components/dialog-add-version.tsx

```diff
@@ -116,15 +131,22 @@ export class AddVersionDialog extends React.Component<
   }
 
   get buttons() {
-    const canSubmit = this.state.isValidElectron && this.state.isValidVersion;
+    const {
+      isValidElectron,
+      isValidVersion,
+      existingLocalVersion,
+    } = this.state;
+    const canSubmit = isValidElectron && isValidVersion;
+    // swap to old local electron version if the user adds a new one with the same path
+    const shouldSetToPreviousVersion = isValidElectron && existingLocalVersion;
```

Comment:
I know I had a run at reviewing this section before with the `disabled={foo}` line but this whole section is still hard for me to scan, so I'm going to try meddling again. :slightly_smiling_face: 

- I think part of it is that we can still submit even when `canSubmit` is false since this dialog now has a dual role.
- Another part is that `shouldSetToPreviousVersion` is a little unclear, we could rename it to draw a line between it and `existingLocalVersion` more clearly.
- Thirdly, the `disabled={foo}` logic, even revised, is still kind of hard to scan.

```suggestion
    const canAdd = isValidElectron && isValidVersion && !existingLocalVersion;
    const canSwitch = isValidElectron && existingLocalVersion;
```

To me this is a lot easier to read, e.g. `canAdd` and `canSwitch` are terse and their relationship to each other is easy to understand at-a-glance. And this also makes the `disabled={}` line clearer: `disabled={!canAdd && !canSwitch}`